core/src/banking_stage/consume_worker.rs

```diff
@@ -122,6 +148,460 @@ fn try_drain_iter<T>(work: T, receiver: &Receiver<T>) -> impl Iterator<Item = T>
     std::iter::once(work).chain(receiver.try_iter())
 }
 
+/// Metrics tracking number of packets processed by the consume worker.
+/// These are atomic, and intended to be reported by the scheduling thread
+/// since the consume worker thread is sleeping unless there is work to be
+/// done.
+pub(crate) struct ConsumeWorkerMetrics {
+    id: u32,
+    interval: AtomicInterval,
+    latch: AtomicBool,
+
+    count_metrics: ConsumeWorkerCountMetrics,
+    error_metrics: ConsumeWorkerTransactionErrorMetrics,
+    timing_metrics: ConsumeWorkerTimingMetrics,
+}
+
+impl ConsumeWorkerMetrics {
+    /// Report and reset metrics iff the interval has elapsed and the worker did some work.
+    pub fn maybe_report_and_reset(&self) {
+        const REPORT_INTERVAL_MS: u64 = 1000;
+        if self.interval.should_update(REPORT_INTERVAL_MS)
+            && self.latch.swap(false, Ordering::Relaxed)
+        {
+            self.count_metrics.report_and_reset(self.id);
+            self.timing_metrics.report_and_reset(self.id);
+            self.error_metrics.report_and_reset(self.id);
+        }
+    }
+
+    fn new(id: u32) -> Self {
+        Self {
+            id,
+            interval: AtomicInterval::default(),
+            latch: AtomicBool::new(false),
+            count_metrics: ConsumeWorkerCountMetrics::default(),
+            error_metrics: ConsumeWorkerTransactionErrorMetrics::default(),
+            timing_metrics: ConsumeWorkerTimingMetrics::default(),
+        }
+    }
+
+    fn update_for_consume(
+        &self,
+        ProcessTransactionBatchOutput {
+            cost_model_throttled_transactions_count,
+            cost_model_us,
+            execute_and_commit_transactions_output,
+        }: &ProcessTransactionBatchOutput,
+    ) {
+        self.count_metrics
+            .cost_model_throttled_transactions_count
+            .fetch_add(*cost_model_throttled_transactions_count, Ordering::Relaxed);
```

Comment:
Found it be useful to know if cost throttling was due to block space or account(s). Or those counters are no different from qos or cost_tracker metrics?

Reply:
QoS metrics are collected and reported the exact same way they were before.
It's "lazily" reported in `process_and_record_transactions_with_pre_results` if the slot has changed, so that is done per slot.
Although as I mentioned to you in private last week, I beleive there are some issues with that reporting method 😄

Reply:
right, that per-slot reporting may not be accurate. 

But here, is `cost_model_throttled_transactions_count` different what qos reports?

Reply:
Believe this is the effectively the same albeit reported on an interval.
With the new metrics I was trying to report all aggregate counts/timings that get returned as part of `ExecuteAndCommitTransactionsOutput`. Current BankingStage metrics are a subset of those.